lightning_pose/data/dali.py

```diff
@@ -168,72 +145,56 @@ def __init__(
         self.do_context = do_context
         self.eval_mode = eval_mode
         self.context_sequences_successive = context_sequences_successive
-        self.batch_sampler = 1 # hack to get around DALI-ptl issue
+        self.batch_sampler = 1  # hack to get around DALI-ptl issue
         # call parent
         super().__init__(*args, **kwargs)
 
     def __len__(self):
         return self.num_iters
     
     def _modify_output(
-            self, out
-        ) -> Union[
-                TensorType["sequence_length", 3, "image_height", "image_width"],
-                TensorType["batch", 5, 3, "image_height", "image_width"]
-            ]:
+        self, out
+    ) -> tuple:
+          #-> Union[
+          #      TensorType["sequence_length", 3, "image_height", "image_width"],
+          #      TensorType["batch", 5, 3, "image_height", "image_width"]
+          #  ]:
         """ modify output to be torch tensor. 
         looks different for context and non-context."""
-        if not self.do_context:
-            return torch.tensor(
-                out[0]["x"][0, :, :, :, :],  # should be (sequence_length, 3, H, W)
-                dtype=torch.float,
-            )  # careful: only valid for one sequence, i.e., batch size of 1.
-        else:
+        if self.do_context:
             if self.eval_mode == "predict":
-                return out[0]["x"]
+                # shape (sequence_length, 3, H, W)
+                images = out[0]["x"]
+                # shape (1,)
+                transform = out[0]["transform"]
             else:
-                # train with context. pipeline is like for "base" model. but we reshape images.
-                # assume batch_size=1
+                # train with context. pipeline is like for "base" model, but we reshape images
+                # further down. assume batch_size=1
                 if self.context_sequences_successive:
-                    # same as for base model
-                    raw_out = torch.tensor(
```

Comment:
I see what you're doing below. we don't need that `torch.tensor(..., dtype=torch.float)` anymore?

Reply:
I don't think so...the variable `out` is already a tensor, and we were getting warnings about how that tensor was being accessed. This is how I've been training the latest models, don't think there's a problem.